### PR TITLE
WIP: Implement commands

### PR DESCRIPTION
**Pull request by forki**
paket.dependencies commands:
- [x] Check for Updates
- [x] Add package
- [x] Update
- [x] Install
- [x] Restore
- [x] Simplify

package commands (below paket.dependencies):
- [x] Update package
- [x] Remove package

package commands (below paket.references):
- [x] Update package
- [x] Remove package
- [x] Add package

paket-update-package-vs

### CODE REVIEW

**Comment by hmemcpy** [comment]
I think maybe make that NRE check inside the method, and return null from
it. So that it always sets the current node to null, in case of something
going wrong.

The joys of shared state!

**Comment by forki** [comment]
Yes will do. 

Does it know it's parent node? If not: I could parse the dependencies file location from the node ID

**Comment by hmemcpy** [comment]
There's a property there to extract the parent node, I think. Look at
graphnodeextensions, I think there's an extension method there

File: src/Paket.VisualStudio/PaketErrorPane.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+using Microsoft.VisualStudio.Shell.Interop;
+using System;
+using Microsoft.VisualStudio.Shell;
+using EnvDTE;
+
+namespace Paket.VisualStudio
+{
+    internal class PaketErrorPane
+    {
+        private static ErrorListProvider vsbaseWarningProvider;
```
Comment (sharwell) [inline src/Paket.VisualStudio/PaketErrorPane.cs]: :bulb: You should rename this :wink:

Comment (forki) [inline src/Paket.VisualStudio/PaketErrorPane.cs]: good point.

Comment (forki) [inline src/Paket.VisualStudio/PaketErrorPane.cs]: any ideas why the error help doesn't work?

Comment (sharwell) [inline src/Paket.VisualStudio/PaketErrorPane.cs]: @forki I'm not sure what you mean.

Comment (forki) [inline src/Paket.VisualStudio/PaketErrorPane.cs]: If I press F1 on the error it should open a webpage, right?

2015-05-12 20:35 GMT+02:00 Sam Harwell notifications@github.com:

Comment (forki) [inline src/Paket.VisualStudio/PaketErrorPane.cs]: Ah we implemented "Navigate", not help

**Pull request merged**